src/components/transport_manager/src/transport_adapter/threaded_socket_connection.cc

```diff
@@ -65,7 +65,7 @@ ThreadedSocketConnection::ThreadedSocketConnection(
 ThreadedSocketConnection::~ThreadedSocketConnection() {
   SDL_AUTO_TRACE();
   Disconnect();
-  thread_->join();
+  thread_->join(true);
```

Comment:
@LuxoftAKutsan, fixed in [fec205f](https://github.com/smartdevicelink/sdl_core/pull/707/commits/fec205ff042a725c2e337f50ebae43cd9a7b22c9)